src/main/java/duke/Deadline.java

```diff
@@ -0,0 +1,40 @@
+package duke;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+
+public class Deadline extends Task {
+
+    protected LocalDateTime by; 
+
+    /**
+     * Deadline task constructor
+     *
+     * @param description description of command
+     * @param by date and time deadline is due
+     * @param isDone status of task
+     */
+    public Deadline(String description, LocalDateTime by, boolean isDone) {
+        super(description,isDone);
+        this.by = by;
+    }
+
+    /**
+     * Parse date into MMM dd yyyy format
+     *
+     * @return date in MMM dd yyyy hh:mm a format
```

Comment:
@return's description can probably be changed to something simpler so as to avoid repetition